internal/services/cdn/frontdoorruleactions/cdn_frontdoor_rule_actions.go

```diff
@@ -207,49 +208,104 @@ func ExpandCdnFrontDoorUrlRewriteAction(input []interface{}) (*[]cdn.BasicDelive
 
 func ExpandCdnFrontDoorRouteConfigurationOverrideAction(input []interface{}) (*[]cdn.BasicDeliveryRuleAction, error) {
 	output := make([]cdn.BasicDeliveryRuleAction, 0)
-
 	m := InitializeCdnFrontDoorActionMappings()
 
 	for _, v := range input {
 		item := v.(map[string]interface{})
 
-		originGroupOverride := &cdn.OriginGroupOverride{
-			OriginGroup: &cdn.ResourceReference{
-				ID: utils.String(item["cdn_frontdoor_origin_group_id"].(string)),
-			},
-			ForwardingProtocol: cdn.ForwardingProtocol(item["forwarding_protocol"].(string)),
-		}
-
+		var originGroupOverride cdn.OriginGroupOverride
+		var cacheConfiguration cdn.CacheConfiguration
+		originGroupIdRaw := item["cdn_frontdoor_origin_group_id"].(string)
+		protocol := item["forwarding_protocol"].(string)
+		cacheBehavior := item["cache_behavior"].(string)
 		compressionEnabled := cdn.RuleIsCompressionEnabledEnabled
+
 		if !item["compression_enabled"].(bool) {
 			compressionEnabled = cdn.RuleIsCompressionEnabledDisabled
 		}
 
-		cacheConfiguration := &cdn.CacheConfiguration{
-			QueryStringCachingBehavior: cdn.RuleQueryStringCachingBehavior(item["query_string_caching_behavior"].(string)),
-			QueryParameters:            expandStringSliceToCsvFormat(item["query_string_parameters"].([]interface{})),
-			IsCompressionEnabled:       compressionEnabled,
-			CacheBehavior:              cdn.RuleCacheBehavior(item["cache_behavior"].(string)),
-			CacheDuration:              utils.String(item["cache_duration"].(string)),
+		// set the default value for forwarding protocol to avoid a breaking change...
+		if protocol == "" {
+			protocol = string(cdn.ForwardingProtocolMatchRequest)
 		}
```

Comment:
this'll want to be changed in v4.0, so could we feature-flag this?

Reply:
I personally believe that we should keep the default values in v4.0 as well since the service will return a 400 bad request error if the fields are not in the create/update call. I can feature-flag them if you would like, but I think that would lead to a bad user experience in v4.0 if we remove them.